packages/browser-wallet/src/popup/pages/Recovery/RecoveryMain.tsx

```diff
@@ -25,21 +25,21 @@ export default function RecoveryMain() {
     const onError = useCallback(
         (reason: string) =>
             navigate(absoluteRoutes.prompt.recovery.path, {
-                state: { status: BackgroundResponseStatus.Error, reason },
+                state: { payload: { status: BackgroundResponseStatus.Error, reason } },
             }),
         []
     );
 
     useEffect(() => {
-        if (runRecovery && !providers.length) {
+        if (runRecovery && !providers.length && seedPhrase.state === 'hasData') {
```

Comment:
I don't get why you need to wait for the seed phrase to be present to get the identity providers. They seem independent.

Reply:
Yeah, that was to avoid crashing the wallet when this failed.